core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -1463,4 +1465,74 @@ abstract class Bundle extends Record {
     *   the fields in the order they were defined
     */
   override def toPrintable: Printable = toPrintableHelper(_elements.toList.reverse)
+
+  /** An optional FIRRTL type alias name to give to this bundle. If overrided with a Some, for instance Some("UserBundle"),
+    * this causes emission of FIRRTL statements that declare that name for this bundle type:
+    *
+    * ```
+    * type UserBundle = { ... }
+    * ```
+    *
+    * This is used as a strong hint for the generated type alias: steps like sanitization and disambiguation
+    * may change the resulting alias by necessity, so there is no certain guarantee that the desired name will show up in
+    * the generated FIRRTL.
+    */
+  def aliasName: Option[BundleAlias] = None
+
+  // The final sanitized and disambiguated alias for this bundle, generated when aliasName is a non-empty Option.
+  // This is important if sanitization and disambiguation results in a changed alias,
+  // as the sanitized name no longer matches the user-specified alias.
+  private[chisel3] var finalizedAlias: Option[String] = None
+
+  private[chisel3] override def bind(target: Binding, parentDirection: SpecifiedDirection): Unit = {
+    super.bind(target, parentDirection)
+
+    aliasName.map(alias => {
+      // If the aliased bundle is coerced and it has flipped signals, then they must be stripped
+      val isFlipped = DataMirror
+        .collectMembers(this) { case d: Data if d.passiveDirection == SpecifiedDirection.Flip => d }
+        .toSeq
+        .nonEmpty
+      val isCoerced = direction match {
+        case ActualDirection.Input | ActualDirection.Output => true
+        case other                                          => false
+      }
+      val isStripped = isCoerced && isFlipped
+
+      // Don't emit an alias for a stripped type
+      val sourceInfo = alias.info
+      val candidateAlias = s"${sanitize(alias.id)}${if (isStripped) "_stripped" else ""}"
```

Comment:
there was a thought that we should make this `_stripped` adding something user configurable, so something like `def aliasNameIfStripped` that is also optional